scout/commands/serve.py

```diff
@@ -19,10 +21,16 @@
 @with_appcontext
 def serve(host, port, debug, livereload, test):
     """Start the web server."""
-    if test:
-        if current_app.config.get("MONGO_DATABASE"):
+
+    # Verify the database connectivity before launching the app
+    mongo_client = current_app.config.get("MONGO_DATABASE")._Database__client
```

Comment:
Hm, isn't that a little dangerous if you dont check for .get("MONGO_DATABASE") so that its not just `None._Database__client`?

Reply:
MONGO_DATABASE will never be null because even if you write nonsense in the connection string, the MongoClient will be created anyway, but it will contain nonsensical params (try!). The validity of the client is checked in the try block immediately after